Hercules-DJControl-Inpulse-200-script.js

```diff
@@ -0,0 +1,218 @@
+        // Released the wheel.
+        engine.scratchDisable(1);
+    }
+};
+
+DJCi200.wheelTouchDB = function(channel, control, value, status, group) {
+    channel = channel + 2;
+    if (value > 0 && (engine.getValue("[Channel2]", "play") !== 1 || DJCi200.scratchButtonStateDB)) {
+        // Touching the wheel.
+        var alpha = 1.0 / 8;
+        var beta = alpha / 32;
+        engine.scratchEnable(2, 400, 33 + 1 / 3, alpha, beta);
+    } else {
+        // Released the wheel.
+        engine.scratchDisable(2);
+    }
+};
+
+
+
+// Using the top of wheel for scratching (Vinyl button On) and bending (Vinyl button Off)
+DJCi200.scratchWheelDA = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    if (engine.isScratching(1)) {
+        engine.scratchTick(1, newValue); // Scratch!
+    } else {
+        engine.setValue('[Channel' + 1 + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+DJCi200.scratchWheelDB = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    // In either case, register the movement
+    if (engine.isScratching(2)) {
+        engine.scratchTick(2, newValue); // Scratch!
+    } else {
+        engine.setValue('[Channel' + 2 + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+// Using the side of wheel for the bending
+DJCi200.bendWheelDA = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    {
+        engine.setValue('[Channel' + 1 + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+
+// The wheel that actually controls the bending
+DJCi200.bendWheelDB = function(channel, control, value, status, group) {
+
+    // A: For a control that centers on 0:
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    {
+        engine.setValue('[Channel' + 2 + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+// FX Pad #4 setting for scratching
+DJCi200.scratchPadDA = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    if (engine.isScratching(1)) {
+        engine.scratchTick(1, newValue); // Scratch!
+    }
+
+};
+
+DJCi200.scratchPadDB = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    // In either case, register the movement
+    if (engine.isScratching(2)) {
+        engine.scratchTick(2, newValue); // Scratch!
+    }
+};
+
+
+
+DJCi200.shutdown = function() {
+
+    midi.sendShortMsg(0xB0, 0x7F, 0x00);
+};
```

Comment:
Please add a newline at end of file. Most editors do this automatically, which would pollute patch diffs later on. So better at it now ;-)